### PR TITLE
solution

### PR DESCRIPTION
**Pull request by MaxSchmide**
[DEMO LINK](https://maxschmide.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Review by ollavka — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -29,128 +69,95 @@ export const App: React.FC = () => {
...
+
+            {/* don't show this button if there are no completed todos */}
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              disabled={completedTodos.length === 0}
+            >
+              Clear completed
+            </button>
+          </footer>
```
Comment (anstsot) [inline src/App.tsx:142]: create separate `Footer` component

File: src/App.tsx [hunk 2]
```diff
@@ -29,128 +69,95 @@ export const App: React.FC = () => {
...
-
-            <span className="todo__title">Todo is being saved now</span>
-            <button type="button" className="todo__remove">×</button>
-
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
+          {todos.map(todo => (
+            <div
```
Comment (anstsot) [inline src/App.tsx:73]: create separate `TodoItem` component

**Review by anstsot — CHANGES_REQUESTED** [review]
Create the component for errors as said in the task, please
image

File: src/App.tsx [hunk 3]
```diff
@@ -1,10 +1,54 @@
...
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((res) => {
+        setTodos(res);
+        setTodosContainer(res);
+        setCompletedTodos(res.filter(todo => todo.completed));
+      })
+      .catch((error) => {
+        throw error;
```
Comment (anstsot) [inline src/App.tsx:48]:
```suggestion
        setErrorMessage(error);
```

**Review by anstsot — CHANGES_REQUESTED** [review]
Try to set error message on load

File: src/App.tsx [hunk 4]
```diff
@@ -1,10 +1,54 @@
...
+import TodoItem from './components/TodoItem';
+import { Todo, TodoStatus } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 11033;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosContainer, setTodosContainer] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
```
Comment (anstsot) [inline src/App.tsx:16]: the variable `completedTodos` can be calculated based on `todosContainer` by using `useMemo` hook

File: src/App.tsx [hunk 5]
```diff
@@ -1,10 +1,54 @@
...
+import ErrorToast from './components/ErrorToast';
+import TodoFooter from './components/TodoFooter';
+import TodoItem from './components/TodoItem';
+import { Todo, TodoStatus } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 11033;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
```
Comment (anstsot) [inline src/App.tsx:14]: the variable `todos` can be calculated based on `todosContainer` and `selectedStatus` by using `useMemo` hook

**Review by anstsot — CHANGES_REQUESTED** [review]
Let's add few more fixes and reduce the number of state variables

File: src/App.tsx [hunk 6]
```diff
@@ -29,128 +76,27 @@ export const App: React.FC = () => {
...
-            <button type="button" className="todo__remove">×</button>
-
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+          {todos.map(todo => (
+            <TodoItem todo={todo} />
```
Comment (anstsot) [inline src/App.tsx:80]: don't forget about `key`

**Review by anstsot — CHANGES_REQUESTED** [review]
There is still a bug with an error showing. maybe you forgot to redeploy?

**Review by anstsot — APPROVED** [review]

**Pull request closed without merging**